specification/draft-2/README.md

```diff
@@ -0,0 +1,687 @@
+TODO: Intro
+
+ - CWL specifies a way of describing runnable command line tools and workflows.
+ - CWL descriptions are serialized as either YAML or JSON.
+ - A CWL description contains a formal declaration of the inputs and outputs of described process (using Apache Avro schema).
+
+
+# Core specification
+
+CWL specification comes in two parts:
+
+ - Core spec allows description of basic tools and workflows
+ - Extensions add features that enable describing more complex tools and workflows
+
+If using an extension feature, it needs to be explicitly listed in the requirements section
+of the description document. Vendors might also add features incompatible with rest of CWL -
+in this case, they should be explicitly listed as requirements with full URI as value of `class` property.
+
+
+## Command line tools
+
+Basic tool example:
+
+```yaml
+class: CommandLineTool
+label: Grep
+description: Filter lines of text file by regular expression.
+inputs:
+  - id: pattern
+    type: string
+    binding:
+      position: 1
+  - id: in_file
+    type: File
+    streamable: true
+    binding:
+      position: 2
+outputs:
+  - id: out_file
+    type: File
+    streamable: true
+    binding:
+      glob: output.txt
+baseCmd: grep
+stdout: output.txt
```

Comment:
There are a few reasons I'd like to have this.
a) It's shorter, and it makes the connection between stdout and the actual output parameter explicit
b) To address your point about determining the file name, you're right, although in my implementation it determines the file name from the parameter name.
c) It provides feature pairity with stdin, which actually a much more important use case because you can't predict the filename in advance and our previous approach to doing this relied on expressions :-(
```yaml
inputs:
  - id: "#inputfile"
    type: File
outputs:
  - id: "#outputfile"
    type: File
stdin: {ref: "#inputfile"}
stdout: {ref: "#outputtfile"}
```